src/main/cli/process.ts

```diff
@@ -0,0 +1,111 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as debug_ from "debug";
+import { app } from "electron";
+import * as path from "path";
+import { _PACKAGING } from "readium-desktop/preprocessor-directives";
+import * as yargs from "yargs";
+import { cli_, cliImport, cliOpds, cliRead } from "./commandLine";
+
+declare const __APP_NAME__: string;
+
+// Logger
+const debug = debug_("readium-desktop:cli");
+
+export function cli(mainFct: () => void) {
+
+    yargs
+        .scriptName(__APP_NAME__)
+        .usage("$0 <cmd> [args]")
+        .command("opds <title> <url>",
+            "import opds feed",
+            (y) =>
+                y.positional("title", {
+                    describe: "title opds feed",
+                    type: "string",
+                })
+                .positional("url", {
+                    describe: "url opds feed",
+                    type: "string",
+                })
+            ,
+            (argv) => {
+                const promise = cliOpds(argv.title, argv.url);
+                promise.then((isValid) => {
+                    if (isValid) {
+                        app.exit(0);
+                        return ;
+                    }
+                    app.exit(1);
+                }).catch(() => {
+                    app.exit(1);
+                });
+            },
+        )
+        .command("import <path>",
+            "import epub or lpcl file",
+            (y) =>
+                y.positional("path", {
+                    describe: "path of your publication",
+                    type: "string",
+                    coerce: (arg) => path.resolve(arg),
+                })
+            ,
+            (argv) => {
+                const promise = cliImport(argv.path);
+                promise.then((isValid) => {
+                    if (isValid) {
+                        app.exit(0);
+                        return ;
+                    }
+                    app.exit(1);
+                }).catch(() => {
+                    app.exit(1);
+                });
+            },
+        )
+        .command("read <title>",
+            "searches already-imported publications with the provided TITLE, and opens the reader with the first match",
+            (y) =>
+                y.positional("title", {
+                    describe: "title of your publication",
+                    type: "string",
+                })
+            ,
+            (argv) => {
+                mainFct();
+                app.on("will-finish-launching", async () => {
+                    await cliRead(argv.title);
```

Comment:
Tell me what you think about the UX:
if the `cliRead` promise returns false, or if it is rejected (`try`/`catch` or `.catch()`), I think we should `app.quit()` gracefully (not `app.exit()`) after a few milliseconds
Rationale: the user intentionally invoked the CLI with `read <title>`, so it seems unnecessary to keep the blank browser window opened in case of failure. Instead: show the error message in the console, and abort gracefully.

Reply:
It not quit with app.exit. The application is just launched on the library and not on the reader.
I will update the code with a try-catch and debug messages

Reply:
So just to clarify: your personal opinion about the UX is that: if the user invokes `thorium.exe TITLE` from the console (or from an OS shortcut), and the TITLE does not match any publication => the application should remain open (library view)?

If so, I just want to clarify that I personally feel the app should shut down gracefully, with `process.stderr.print` and either `app.exit(code)` (abort), or `app.quit()` (clean exit, just in case some destroy/teardown functionality depends on the Electron `quit` events)

What do you think?

Reply:
an error message in react can be displayed to the user .. but for me, in any case, it must leave.
When a user double-clicks on an epub, he will not see thorium at all.

the launch principle of thorium is the opening of the library then the reader.

Reply:
> Otherwise, when a user double-clicks on an epub, he will not see thorium at all.

Right, that makes sense. So for example `thorium.exe TITLE` could be used via an icon shortcut on the user's desktop, it is not necessarily a command line / shell invocation. In which case, the user must be able to get some GUI feedback, instead of just `stderr`. What about Electron's own built-in `dialog.showErrorBox()`?
https://electronjs.org/docs/api/dialog#dialogshowerrorboxtitle-content

Reply:
done